File: php-demo/ValidateDemo.php
<?php
/** 产品密钥ID，产品标识 */
define("SECRETID", "your_secret_id");
/** 产品私有密钥，服务端生成签名信息使用，请严格保管，避免泄露 */
define("SECRETKEY", "your_secret_key");
/** 业务ID，易盾根据产品业务特点分配 */
define("BUSINESSID", "your_business_id");
/** 易盾短信服务发送接口地址 */
define("API_URL", "https://ye.dun.163yun.com/v1/check");
/** api version */
define("VERSION", "v1");
/** API timeout*/
define("API_TIMEOUT", 5);
/** php内部使用的字符串编码 */
define("INTERNAL_STRING_CHARSET", "auto");
/**
 * 计算参数签名
 * $params 请求参数
 * $secretKey secretKey
 */
function gen_signature($secretKey, $params)
{
    ksort($params);
    $buff = "";
    foreach ($params as $key => $value) {
        if ($value !== null) {
            $buff .= $key;
            $buff .= $value;
        }
    }
    $buff .= $secretKey;
    return md5($buff);
}
/**
 * 将输入数据的编码统一转换成utf8
 * @params 输入的参数
 */
function toUtf8($params)
{
    $utf8s = array();
    foreach ($params as $key => $value) {
        $utf8s[$key] = is_string($value) ? mb_convert_encoding($value, "utf8", INTERNAL_STRING_CHARSET) : $value;
    }
    return $utf8s;
}
/**
 * 易盾本机验证在线检测请求接口简单封装
 * $params 请求参数
 */
function check($params)
{
    $params["secretId"] = SECRETID;
    $params["businessId"] = BUSINESSID;
    $params["version"] = VERSION;
    $params["timestamp"] = sprintf("%d", round(microtime(true) * 1000));
    // time in milliseconds
    $params["nonce"] = substr(md5(time()), 0, 32);
    // random int
    $params = toUtf8($params);
    $params["signature"] = gen_signature(SECRETKEY, $params);
    $options = array('http' => array(
        'header' => "Content-type: application/x-www-form-urlencoded\r\n",
        'method' => 'POST',
        'timeout' => API_TIMEOUT,
        // read timeout in seconds
        'content' => http_build_query($params),
    ));
    $context = stream_context_create($options);
    $result = file_get_contents(API_URL, false, $context);
    if ($result === FALSE) {
        return array("code" => 500, "msg" => "file_get_contents failed.");
    } else {
        return json_decode($result, true);
    }
}
// 简单测试
function main()
{
    $params = array(
        // 运营商预取号获取到的token
        "accessToken" => "xx",
        // 易盾返回的token
        "token" => "xxx",
        "phone" => "xxx"
    );
    $ret = check($params);
    var_dump($ret);
    if ($ret["code"] == 200) {
        $result = $ret["result"];
        if ($result == 1) {
            // 1通过
            // 执行后续业务逻辑处理
        } else if ($result == 2) {
            // 2不通过
            // 建议进行二次验证,例如短信验证码
        } else {
            // 3无法确定
            // 建议进行二次验证,例如短信验证码
        }
    } else {
        var_dump($ret);
        // error handler
        // 建议进行二次验证,例如短信验证码
    }
}
main();

File: go-demo/ValidateDemo.go
package main

import (
	"crypto/md5"
	"encoding/hex"
	"encoding/json"
	"fmt"
	simplejson "github.com/bitly/go-simplejson"
	"io/ioutil"
	"net/http"
	"net/url"
	"sort"
	"strconv"
	"strings"
	"time"
)

const (
	apiUrl     = "https://ye.dun.163yun.com/v1/check"  //本机认证服务身份证实人认证在线检测接口地址
	version    = "v1"
	secretId   = "your_secret_id"   //产品密钥ID，产品标识
	secretKey  = "your_secret_key"  //产品私有密钥，服务端生成签名信息使用，请严格保管，避免泄露
	businessId = "your_business_id" //业务ID，易盾根据产品业务特点分配
)

//请求易盾接口
func check(params url.Values) *simplejson.Json {
	params["secretId"] = []string{secretId}
	params["businessId"] = []string{businessId}
	params["version"] = []string{version}
	params["timestamp"] = []string{strconv.FormatInt(time.Now().UnixNano()/1000000, 10)}
	params["nonce"] = []string{string(make([]byte, 32))}    //32位随机字符串
	params["signature"] = []string{gen_signature(params)}

	resp, err := http.Post(apiUrl, "application/x-www-form-urlencoded", strings.NewReader(params.Encode()))

	if err != nil {
		fmt.Println("调用API接口失败:", err)
		return nil
	}

	defer resp.Body.Close()

	contents, _ := ioutil.ReadAll(resp.Body)
	result, _ := simplejson.NewJson(contents)
	return result
}


//生成签名信息
func gen_signature(params url.Values) string {
	var paramStr string
	keys := make([]string, 0, len(params))
	for k := range params {
		keys = append(keys, k)
	}
	sort.Strings(keys)
	for _, key := range keys {
		paramStr += key + params[key][0]
	}
	paramStr += secretKey
	md5Reader := md5.New()
	md5Reader.Write([]byte(paramStr))
	return hex.EncodeToString(md5Reader.Sum(nil))
}

func main() {
	params := url.Values{
	            //phone为手机号
	            "phone":   []string{"12345678912"},
    	        //token为易盾返回的token
        		"token":   []string{"123456"},
        		//accessToken为运营商预取号获取到的token
        		"accessToken": []string{"123456"},
    }

	ret := check(params)

	code, _ := ret.Get("code").Int()

	if code == 200 {
         result, _ := ret.Get("result").Int()
         if result == 1 {
         	   fmt.Printf("通过,执行后续业务逻辑处理!")
         } else if result == 2 {
               fmt.Printf("不通过,建议进行二次验证,例如短信验证码!")
         } else if result == 3 {
               fmt.Printf("无法确定,建议进行二次验证,例如短信验证码!")
         }
	} else {
		fmt.Printf("建议进行二次验证,例如短信验证码!")
	}
}
